src/commands/sign-requests/create.js

```diff
@@ -1,56 +1,196 @@
+			};
+
+			for (const part of input.split(',')) {
+				const [
+					key,
+					value
+				] = part.split('=');
+
+				switch (key) {
+				case 'email':
+					signer.email = value;
+					break;
+
+				case 'role':
+					if (!ALLOWED_SIGNER_ROLES.includes(value)) {
+						throw new BoxCLIError(
+							`Invalid value for role property of signer: ${value}. Expecting one of: ${ALLOWED_SIGNER_ROLES.join(
+								', '
+							)}.`
+						);
+					}
+					signer.role = value;
+					break;
+
+				case 'is_in_person':
+					if (value !== '0' && value !== '1') {
+						throw new BoxCLIError(
+							`Invalid value for is_in_person property of signer: ${value}. Expecting either 0 or 1.`
+						);
+					}
+					signer.is_in_person = value === '1';
+					break;
+
+				case 'order':
+					signer.order = value;
+					break;
+
+				case 'embed_url_external_user_id':
+					signer.embed_url_external_user_id = value;
+					break;
+
+				default:
+					throw new BoxCLIError(`Unknown proprty for signer: ${key}`);
+				}
+			}
+
+			return signer;
+		},
 	}),
 	'source-files': flags.string({
-		char: 'f',
 		required: true,
 		description:
 			'Comma separated list of files to create a signing document from. This is currently limited to one file, e.g. 12345',
+		parse: input => input.split(',').map(id => ({
+			type: 'file',
+			id,
+		})),
 	}),
 	'parent-folder': flags.string({
-		char: 'p',
 		required: true,
 		description:
 			'The destination folder to place final, signed document and signing log',
+		parse: input => ({
+			type: 'folder',
+			id: input,
+		}),
+	}),
+	'document-preparation-needed': flags.boolean({
+		description:
+			'Indicates if the sender should receive a `prepare_url` in the response to complete document preparation via UI.',
+		allowNo: true,
+	}),
+	'text-signatures-enabled': flags.boolean({
+		description: 'Disables the usage of signatures generated by typing (text)',
+		allowNo: true,
+	}),
+	'email-subject': flags.string({
+		description:
+			'Subject of sign request email. This is cleaned by sign request. If this field is not passed, a default subject will be used.',
+	}),
+	'email-message': flags.string({
+		description:
+			'Message to include in sign request email. The field is cleaned through sanitization of specific characters. However, some html tags are allowed. Links included in the message are also converted to hyperlinks in the email. The message may contain the following html tags including `a`, `abbr`, `acronym`, `b`, `blockquote`, `code`, `em`, `i`, `ul`, `li`, `ol`, and `strong`. Be aware that when the text to html ratio is too high, the email may end up in spam filters. Custom styles on these tags are not allowed. If this field is not passed, a default message will be used.',
+	}),
+	'reminders-enabled': flags.boolean({
+		description:
+			'Reminds signers to sign a document on day 3, 8, 13 and 18. Reminders are only sent to outstanding signers.',
+		allowNo: true,
+	}),
+	'prefill-tag': flags.string({
```

Comment:
It should be plural - 'tags', right?

Reply:
I made it singular because using this flag you can set one tag. The flag can be used multiple times.